### PR TITLE
ToolkitDemo more controls

### PR DESCRIPTION
**Pull request by rehan-saleh**
### Added following controls
- Borderless Editor
- Borderless Picker
- Material Button
- Material Picker
- Lowercase converter

Issue #70

### CODE REVIEW

File: demo/ToolkitDemo.iOS/ToolkitDemo.iOS.csproj [hunk 1]
```diff
@@ -195,6 +195,9 @@
     <Reference Include="Xamarin.iOS" />
   </ItemGroup>
   <ItemGroup>
+	<PackageReference Include="Microsoft.Identity.Client">
+      <Version>4.4.0</Version>
+    </PackageReference>
```
Comment (dansiegel) [inline demo/ToolkitDemo.iOS/ToolkitDemo.iOS.csproj]: please ensure the Package Reference follows the same style as the other references... That said it should be restored transitively from the AAD project and even then it shouldn't be part of this PR as you haven't included anything relating to Authentication.

File: demo/ToolkitDemo.Android/Properties/AndroidManifest.xml [hunk 2]
```diff
@@ -2,5 +2,15 @@
...
+	<application android:label="ToolkitDemo.Android" android:icon="@mipmap/icon">
+		<activity android:name="microsoft.identity.client.BrowserTabActivity">
+			<intent-filter>
+				<action android:name="android.intent.action.VIEW" />
+				<category android:name="android.intent.category.DEFAULT" />
+				<category android:name="android.intent.category.BROWSABLE" />
+				<data android:scheme="msal{REPLACEME}" android:host="auth" />
+			</intent-filter>
+		</activity>
+	</application>
```
Comment (dansiegel) [inline demo/ToolkitDemo.Android/Properties/AndroidManifest.xml]: Why is this change here? There is nothing in this PR that uses the MSAL library. This is completely careless.

File: demo/ToolkitDemo.iOS/AppDelegate.cs [hunk 3]
```diff
@@ -20,9 +21,19 @@ public override bool FinishedLaunching(UIApplication app, NSDictionary options)
...
+            LoadApplication(new App(this));
 
             return base.FinishedLaunching(app, options);
         }
+
+        public override bool OpenUrl(UIApplication app, NSUrl url, NSDictionary options)
+        {
+            AuthenticationContinuationHelper.SetAuthenticationContinuationEventArgs(url);
+            return true;
+        }
```
Comment (dansiegel) [inline demo/ToolkitDemo.iOS/AppDelegate.cs]: Why is this change here? There is nothing in this PR that uses the MSAL library. This is completely careless.

File: demo/ToolkitDemo.iOS/Info.plist [hunk 4]
```diff
@@ -28,6 +28,19 @@
...
+			<key>CFBundleTypeRole</key>
+			<string>Editor</string>
+			<key>CFBundleURLName</key>
+			<string>com.avantipoint.toolkitdemo</string>
+			<key>CFBundleURLSchemes</key>
+			<array>
+				<string>msal$AzureADClientId$</string>
+			</array>
+		</dict>
+	</array>
```
Comment (dansiegel) [inline demo/ToolkitDemo.iOS/Info.plist]: Why is this change here? There is nothing in this PR that uses the MSAL library. This is completely careless.

File: demo/ToolkitDemo/Views/MaterialPickerPage.xaml [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+<ContentPage xmlns="http://xamarin.com/schemas/2014/forms"
+             xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
+             xmlns:prism="http://prismlibrary.com"
+             prism:ViewModelLocator.AutowireViewModel="True"
+             x:Class="ToolkitDemo.Views.MaterialPickerPage"
+             xmlns:toolkit="clr-namespace:AP.MobileToolkit.Controls;assembly=AP.MobileToolkit"
+             Title="Material Picker">
+    <ContentPage.ToolbarItems>
+        <ToolbarItem Order="Primary" Text="Code Behind" Priority="0" Command="{Binding ShowCodeBehind}" />
+    </ContentPage.ToolbarItems>
```
Comment (dansiegel) [inline demo/ToolkitDemo/Views/MaterialPickerPage.xaml]: Please tell me what you think of this code...

File: demo/ToolkitDemo.iOS/Assets.xcassets/AppIcon.appiconset/Contents.json [hunk 6]
```diff
@@ -1,182 +1,117 @@
...
+      "scale": "2x",
+      "size": "83.5x83.5",
+      "idiom": "ipad",
+      "filename": "Icon-App-83.5x83.5@2x.png"
+    },
+    {
+      "scale": "1x",
+      "size": "1024x1024",
+      "idiom": "ios-marketing",
+      "filename": "ItunesArtwork@2x.png"
```
[the remark this answers is not in the document]
Comment (rehan-saleh) [reply demo/ToolkitDemo.iOS/Assets.xcassets/AppIcon.appiconset/Contents.json]: While I was trying to fix ImageEntryPage I added AvantiPoint icon in assets to test if it makes any difference.

File: demo/ToolkitDemo.iOS/AppDelegate.cs [hunk 7]
```diff
@@ -9,7 +10,7 @@ namespace ToolkitDemo.iOS
     // User Interface of the application, as well as listening (and optionally responding) to
     // application events from iOS.
     [Register("AppDelegate")]
-    public partial class AppDelegate : global::Xamarin.Forms.Platform.iOS.FormsApplicationDelegate
+    public partial class AppDelegate : global::Xamarin.Forms.Platform.iOS.FormsApplicationDelegate, IPlatformInitializer
```
[the remark this answers is not in the document]
Comment (rehan-saleh) [reply demo/ToolkitDemo.iOS/AppDelegate.cs]: This was required for AAD demo. I've removed that from this PR and also removed this code.

File: demo/ToolkitDemo.iOS/AppDelegate.cs [hunk 8]
```diff
@@ -20,9 +21,19 @@ public override bool FinishedLaunching(UIApplication app, NSDictionary options)
         {
             global::Xamarin.Forms.Forms.Init();
 
-            LoadApplication(new App());
+            LoadApplication(new App(this));
```
[the remark this answers is not in the document]
Comment (rehan-saleh) [reply demo/ToolkitDemo.iOS/AppDelegate.cs]: This was required for AAD demo. I've removed that from this PR and also removed this code.

File: demo/ToolkitDemo.iOS/AppDelegate.cs [hunk 9]
```diff
@@ -20,9 +21,19 @@ public override bool FinishedLaunching(UIApplication app, NSDictionary options)
...
+
+        public override bool OpenUrl(UIApplication app, NSUrl url, NSDictionary options)
+        {
+            AuthenticationContinuationHelper.SetAuthenticationContinuationEventArgs(url);
+            return true;
+        }
+
+        public void RegisterTypes(IContainerRegistry containerRegistry)
+        {
+        }
```
[the remark this answers is not in the document]
Comment (rehan-saleh) [reply demo/ToolkitDemo.iOS/AppDelegate.cs]: This was included by mistake. I've removed this code from the PR.

File: demo/ToolkitDemo.iOS/ToolkitDemo.iOS.csproj (hunk 1 above)
Comment (rehan-saleh) [reply demo/ToolkitDemo.iOS/ToolkitDemo.iOS.csproj]: I have removed this code from the PR.

File: demo/ToolkitDemo.iOS/Assets.xcassets/AppIcon.appiconset/Contents.json (hunk 6 above)
[the remark this answers is not in the document]
Comment (rehan-saleh) [reply demo/ToolkitDemo.iOS/Assets.xcassets/AppIcon.appiconset/Contents.json]: I have reverted this code in original state.

File: demo/ToolkitDemo.Android/Properties/AndroidManifest.xml (hunk 2 above)
Comment (rehan-saleh) [reply demo/ToolkitDemo.Android/Properties/AndroidManifest.xml]: This was required for AAD demo. I've removed that from this PR and also removed this code.

File: demo/ToolkitDemo.iOS/AppDelegate.cs (hunk 3 above)
Comment (rehan-saleh) [reply demo/ToolkitDemo.iOS/AppDelegate.cs]: This was required for AAD demo. I've removed that from this PR and also removed this code.

File: demo/ToolkitDemo.iOS/Info.plist (hunk 4 above)
Comment (rehan-saleh) [reply demo/ToolkitDemo.iOS/Info.plist]: This was required for AAD demo. I've removed that from this PR and also removed this code.

File: demo/ToolkitDemo/Views/MaterialPickerPage.xaml (hunk 5 above)
Comment (rehan-saleh) [reply demo/ToolkitDemo/Views/MaterialPickerPage.xaml]: I think these lines of code should be generic as they are being repeated. I am looking into it.

**Comment by dansiegel** [comment]
Your changes were not what I explicitly asked you to do... the answer is to fix this with an Interactive Rebase not by adding new commits to back out what should never have been included in the first place
